src/main/kotlin/com/msd/robot/domain/Robot.kt

```diff
@@ -191,4 +191,10 @@ class Robot(
             energyRegen
         }
     }
+
+    fun regenerateHealth(amount: Int) {
+        if (amount + this.health > this.maxHealth) throw HealthFullException("Tried to regenerate $amount health but is already at full health.");
```

Comment:
The robot should still heal up to its max health. This makes it impossible to heal up to 20, if the robot has 17 Health and tries to heal by 4